crates/primitives/src/sol.rs

```diff
@@ -446,3 +447,20 @@ impl SolEncode<'_> for H256 {
         SolBytes(self.0)
     }
 }
+
+// Weight
+impl SolDecode for Weight {
+    type SolType = (u64, u64);
+
+    fn from_sol_type(value: Self::SolType) -> Self {
+        Weight::from_parts(value.0, value.1)
+    }
+}
+
+impl<'a> SolEncode<'a> for Weight {
+    type SolType = (u64, u64);
```

Comment:
Unable to use references due to struct fields being private, requiring use of the getters.

Reply:
`Copy` and primitive types, so performance should be the same anyway